compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java

```diff
@@ -365,6 +366,15 @@ public void visit(BRecordType bRecordType) {
             writeTypeCpIndex(field.type);
         }
 
+        buff.writeInt(bRecordType.originalFields.size());
+        for (BField field : bRecordType.originalFields.values()) {
+            BVarSymbol symbol = field.symbol;
+            buff.writeInt(addStringCPEntry(symbol.name.value));
+            buff.writeLong(symbol.flags);
+            writeMarkdownDocAttachment(buff, field.symbol.markdownDocumentation);
+            writeTypeCpIndex(field.type);
+        }
```

Comment:
Do we really need to be duplicating everything like this? Should be able to just filter out the user specified fields from the fields list and add them to the cached list of user specified fields.

Reply:
Optimized by adding a flag when writing the fields.